options.go

```diff
@@ -29,6 +29,18 @@ func WithMaxKeep(n int) option {
 	}
 }
 
+// WithMaxMessageKeysPerSession specifies the maximum number of message keys per session
+// nolint: golint
+func WithMaxMessageKeysPerSession(n int) option {
```

Comment:
```suggestion
func WithMaxMessageKeysPerSession(n uint) option {
```
so we can avoid the check and type casting.

Reply:
I have kept it as an int as  you suggested to have it as an integer to avoid typecasting when using `len(a) - maxMessageKeys`
let me know if you feel it should be otherwise

Reply:
I see. Let's keep it this way. `option` has `func(s *State) error` signature so it works nice together.